homeassistant/components/reolink/host.py

```diff
@@ -362,57 +362,57 @@ def unregister_webhook(self):
     async def handle_webhook(
         self, hass: HomeAssistant, webhook_id: str, request: Request
     ) -> None:
-        """Shield the incoming webhook callback from cancellation."""
-        shielded_future = asyncio.shield(
-            self._handle_webhook(hass, webhook_id, request)
-        )
+        """Read the incoming webhook from Reolink for inbound messages and schedule processing."""
         _LOGGER.debug("Webhook '%s' called", webhook_id)
-        if not self._webhook_reachable.is_set():
-            self._webhook_reachable.set()
-            ir.async_delete_issue(self._hass, DOMAIN, "webhook_url")
-        await shielded_future
-
-    async def _handle_webhook(
-        self, hass: HomeAssistant, webhook_id: str, request: Request
-    ) -> None:
-        """Handle incoming webhook from Reolink for inbound messages and calls."""
+        data: str | None = None
         try:
             data = await request.text()
```

Comment:
If you want to make it even faster here you could use .read() and decode it in the background task since .text() may try to detect char set which is much slower

Reply:
Done, thanks for the tip!